src/openforms/submissions/form_logic.py

```diff
@@ -93,8 +97,10 @@ def evaluate_form_logic(
 
     updated_step_data = deepcopy(step.data)
     updated_submission_data = deepcopy(data)
+    evaluated_rules = list()
     for rule in rules:
         if jsonLogic(rule.json_logic_trigger, updated_submission_data):
+            evaluated_rules.append({"rule": rule, "trigger": True})
```

Comment:
I'd restructure this quite a bit - now the `if: else` are pretty far apart and ideally we keep the logging code in just a single place:

```suggestion
        trigger_rule = jsonLogic(rule.json_logic_trigger, updated_submission_data)
        evaluated_rules.append({"rule": rule, "trigger": trigger_rule})
        if not trigger_rule:
            continue
        ...  # for action in rules.actions...
```
& this way you can remove the `else` completely a couple lines down.

The `if not trigger_rule: continue` construct also allows us to simplify the code by removing a nesting level